pkg/filesharecontroller/selector/filter.go

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2017 Huawei Technologies Co., Ltd. All Rights Reserved.
```

Comment:
I hope these are common functions, nothing specific to Fileshare .Hope we can reuse pkg/controller/selector/filter.

Reply:
The long term goal is to have an independent module of Selector so that both block and file share service can use them and user should be able to deploy file share service without deploying block service.  Due to time constraint (m-3 deadline is the week of May 6th when all feature code needs to be merged), the code is duplicated in Fileshare Controller in the first phase.  

@Shruthi-1MN and @PravinRanjan10 , please add a TODO in both filter.go and selector.go that refactor work will need to be done to consolidate the selector for block and file share.

Reply:
Sure will do

Reply:
@NajmudheenCT filter can be reused but it is in selector folder, selector we can't use as it is as file share and block has it's own proto buf. Refactor work has to be done.